modules/parsers/parser-hana/src/main/java/custom/HanaProcedureUpdateStatementListener.java

```diff
@@ -0,0 +1,144 @@
+import com.sap.xsk.parser.hana.core.HanaParser.Update_set_clauseContext;
+import com.sap.xsk.parser.hana.core.HanaParser.Update_stmtContext;
+import com.sap.xsk.parser.hana.core.HanaParser.Where_clauseContext;
+import models.FromClauseDefinitionModel;
+import models.JoinClauseDefinitionModel;
+import models.ProcedureDefinitionModel;
+import models.TableReferenceModel;
+import models.UpdateSetClauseDefinitionModel;
+import models.UpdateStatementDefinitionModel;
+import models.WhereClauseDefinitionModel;
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.misc.Interval;
+import org.apache.commons.lang3.StringUtils;
+
+public class HanaProcedureUpdateStatementListener extends HanaBaseListener {
+
+  ProcedureDefinitionModel procedureModel;
+  FromClauseDefinitionModel fromClauseModel;
+  UpdateStatementDefinitionModel updateStatementModel;
+  boolean isUpdateStatementScope = false;
+
+  @Override
+  public void enterCreate_procedure_body(Create_procedure_bodyContext ctx) {
+    String strippedSchema = null;
+    String strippedName = null;
+
+    if (ctx.proc_name() != null) {
+      if (ctx.proc_name().id_expression() != null) {
+        String maybeQuotedName = ctx.proc_name().id_expression().getText();
+        strippedName = StringUtils.strip(maybeQuotedName, "\"");
+      }
+
+      if (ctx.proc_name().schema_name() != null) {
+        String maybeQuotedSchema = ctx.proc_name().schema_name().getText();
+        strippedSchema = StringUtils.strip(maybeQuotedSchema, "\"");
+      }
+    }
+    procedureModel = new ProcedureDefinitionModel(strippedSchema, strippedName);
+  }
+
+  @Override
+  public void enterUpdate_stmt(Update_stmtContext ctx) {
+    isUpdateStatementScope = true;
+    updateStatementModel = new UpdateStatementDefinitionModel();
+    String tableName = ctx.general_table_ref().dml_table_expression_clause().tableview_name().getText();
+    String tableAlias = ctx.general_table_ref().table_alias() != null ? ctx.general_table_ref().table_alias().getText() : null;
+    updateStatementModel.setName(tableName);
+    updateStatementModel.setAlias(tableAlias);
+    updateStatementModel.setRawContent(getStringWithSpaces(ctx));
+    procedureModel.addUpdateStatement(updateStatementModel);
+  }
+
+  @Override
+  public void exitUpdate_stmt(Update_stmtContext ctx) {
+    isUpdateStatementScope = false;
+  }
+
+  @Override
+  public void enterUpdate_set_clause(Update_set_clauseContext ctx) {
+    if (isUpdateStatementScope && ctx.parent instanceof Update_stmtContext) {
+      UpdateSetClauseDefinitionModel updateSetClauseModel = new UpdateSetClauseDefinitionModel();
+      updateSetClauseModel.setRawContent(getStringWithSpaces(ctx));
+      updateStatementModel.setUpdateSetClause(updateSetClauseModel);
+    }
+  }
+
+  @Override
+  public void enterJoin_clause(Join_clauseContext ctx) {
+    if (isUpdateStatementScope && ctx.parent.parent.parent.parent instanceof Update_stmtContext) {
```

Comment:
extract ctx.parent.parent.parent.parent instanceof Update_stmtContext in local variable with proper name, because now it is hard to understand what should be of Update stmt instance.